src/Schema/SqliteSchemaManager.php

```diff
@@ -626,7 +626,7 @@ private function getForeignKeyDetails($table)
                 '#
                     (?:CONSTRAINT\s+(\S+)\s+)?
                     (?:FOREIGN\s+KEY[^)]+\)\s*)?
-                    REFERENCES\s+\S+\s+(?:\([^)]+\))?
+                    REFERENCES\s+\S+\s*(?:\([^)]+\))?
```

Comment:
Note that this pattern is still invalid since the `\S+` expression matches the entire `"users"("id"));` substring of the SQL in the newly added test rendering the rest of the expressions on this line obsolete.

The proper parsing logic would be more complex and might require a real parser since the double quote that terminates the table name acts as a token separator from the SQL syntax perspective but isn't a whitespace charater from the regular expression perspective.

Reply:
Wouldn't that be solved by using `[^\s"]` instead of `\S`?

Reply:
> the double quote that terminates the table name acts as a token separator from the SQL syntax perspective

I guess this isn't entirely accurate since `users(id)` would be also a valid expression but it doesn't contain whitespace or a quote.

At the same time, an identifier can contain quotes, e.g. `"foo""bar"` but the proposed pattern will need an optional leading quote and will match the first half of the identifier.

Reply:
A note for posterity: there is a quite mature parser generator [ANTLR](https://www.antlr.org/) which allows defining grammar in an abstract syntax and generating parsers for a variety of languages including PHP. For example, [Debezium](https://debezium.io/) uses ANTLR-based parsers for parsing arbitrary DDL statements of more database platforms than the DBAL supports.

If we ever get to the point where we'd want to properly support arbitrary identifiers (e.g. `CREATE TABLE "FOREIGN KEY u REFERENCES users(id)"` ← good luck parsing that), we might use ANTLR.